cmake/GaloisTesting.cmake

```diff
@@ -0,0 +1,149 @@
+#     endif ()
+#   endif()
+#   set(NUM_PXNS 2)
+#   set(NUM_CORES 4)
+
+#   pando_add_executable(${TARGET} ${PROJECT_SOURCE_DIR}/test/test_driver.cpp ${SOURCEFILE})
+
+#   if (PANDO_RT_BACKEND STREQUAL "DRVX")
+#     target_link_libraries(${TARGET} PRIVATE
+#       "$<LINK_LIBRARY:WHOLE_ARCHIVE,GTest::gtest>")
+#     set_target_properties(gtest PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+#     # create a dummy executable as a different target but with the same name for ctest to discover the right test programs
+#     add_executable(${TARGET}-drvx ${PROJECT_SOURCE_DIR}/test/test_dummy.cpp)
+#     set_target_properties(${TARGET}-drvx PROPERTIES OUTPUT_NAME ${TARGET})
+
+#     # Robust method to bail on error: create regex variable to track failed tests via stdout/stderr rather than rely on return code of drvx-sst
+#     set(FAIL_REGEX "FAIL_REGULAR_EXPRESSION;Failed;FAIL_REGULAR_EXPRESSION;ERROR;FAIL_REGULAR_EXPRESSION;FAILED;")
+#   else()
+#     target_link_libraries(${TARGET}
+#       PRIVATE
+#         GTest::gtest
+#     )
+#   endif()
+
+#   pando_compiler_options(${TARGET})
+#   pando_compiler_warnings(${TARGET})
+
+#   # use CROSSCOMPILING_EMULATOR to point to the driver script
+#   # it cannot be added to set_target_properties
+#   set_property(TARGET ${TARGET}
+#     PROPERTY CROSSCOMPILING_EMULATOR '${DRIVER_SCRIPT} ${HTHREADS} -n ${NUM_PXNS} -c ${NUM_CORES}'
+#   )
+
+#   if (${PANDO_TEST_DISCOVERY})
+#     gtest_discover_tests(${TARGET}
+#       DISCOVERY_TIMEOUT ${PANDO_TEST_DISCOVERY_TIMEOUT}
+#     )
+#   endif ()
+# endfunction()
+
+# function(pando_add_bin_test TARGET ARGS INPUTFILE OKFILE)
+#   if (NOT PANDO_RT_BACKEND STREQUAL "DRVX")
+#     if (${GASNet_CONDUIT} STREQUAL "smp")
+#       set(DRIVER_SCRIPT ${PROJECT_SOURCE_DIR}/pando-rt/scripts/preprun.sh)
+#     elseif (${GASNet_CONDUIT} STREQUAL "mpi")
+#       set(DRIVER_SCRIPT ${PROJECT_SOURCE_DIR}/pando-rt/scripts/preprun_mpi.sh)
+#     else ()
+#       message(FATAL_ERROR "No runner script for GASNet conduit ${GASNet_CONDUIT}")
+#     endif ()
+
+#     set(NUM_PXNS 2)
+#     set(NUM_CORES 4)
+
+#     add_test(NAME ${TARGET}-${INPUTFILE}-${OKFILE}
+#       COMMAND bash -c "diff -Z <(${DRIVER_SCRIPT} -n ${NUM_PXNS} -c ${NUM_CORES} ${CMAKE_CURRENT_BINARY_DIR}/${TARGET} ${ARGS} ${INPUTFILE}) ${OKFILE}")
+#   else()
+
+#     set(DRIVER_SCRIPT ${PROJECT_SOURCE_DIR}/scripts/run-drv.sh)
+
+#     set(NUM_PXNS 2)
+#     set(NUM_CORES 4)
+#     set(NUM_HTHREADS 8)
+
+#     get_filename_component(FNAME ${TARGET} NAME)
+
+#     add_test(NAME ${TARGET}-${INPUTFILE}-${OKFILE}
+#       COMMAND bash -c "diff -Z <(LAUNCH_DIR=${CMAKE_SOURCE_DIR} ${DRIVER_SCRIPT} -p ${NUM_HTHREADS} -n ${NUM_PXNS} -c ${NUM_CORES} \
+#       ${CMAKE_CURRENT_BINARY_DIR}/lib${FNAME}.so ${ARGS} ${INPUTFILE} \
+#       | grep -v 'memBackendConverter:' |grep -v 'PANDOHammerDrvX:' |grep -v 'Simulation is complete, simulated time:' | tail -n +3) \
+#       ${OKFILE}")
+
+#   endif()
+# endfunction()
```

Comment:
delete this, do not comment it out.

Reply:
fair enough, most of it will be added back in but we can always just go back to the source for it